src/ExtendedMath.f90

```diff
@@ -165,6 +170,56 @@ ELEMENTAL FUNCTION GreatestCommonDivisor(a1,a2) RESULT(b)
 ENDFUNCTION GreatestCommonDivisor
 !
 !-------------------------------------------------------------------------------
+!> @brief solves for the least common multiple of an array of integers
+!> @param u the array of integers to consider
+!> @returns lcm the least common multiple of the array
+!>
+PURE FUNCTION LeastCommonMultiple_A1(u) RESULT(lcm)
+  INTEGER(SIK),INTENT(IN) :: u(:)
+
+  INTEGER(SIK) :: lcm
+  INTEGER(SIK) :: i,a
+
+  lcm = 0
+  IF(SIZE(u) == 0) RETURN
+  IF(ANY(u == 0)) RETURN
+
+  lcm = abs(u(1))
+  IF(ALL(u == u(1))) THEN
+    RETURN
+  ENDIF
+
+  DO i=2,SIZE(u)
+    a = abs(u(i))
+    !In all likelihood there exists an 'a' and 'b' which will cause overflow
+    lcm = (lcm/GreatestCommonDivisor(lcm,a)) * a
+  ENDDO
```

Comment:
So... seems like this entire function can be reduced to this and reduce some redundant logic.  Disagree?
```
lcm = 0
IF(SIZE(u) == 0) RETURN
IF(ANY(u) == 0) RETURN

lcm=a
DO i=2,SIZE(u)
  lcm = LeastCommonMultiple_scalar(ABS(lcm),ABS(u(i))
ENDDO
```

Reply:
Yeah, implemented this less the call to `ABS`. Since its calling the scalar version and the scalar version is agnostic to sign by design, I don't think its necessary.

Reply:
Good point.  Thanks for cleaning up